plan.py

```diff
@@ -55,39 +110,78 @@ def expansion(node):
     # Convert mols to format for prediction
     # If the mol is in the starting set, ignore
     mols = [mol for mol in mols if mol not in starting_mols]
-    fprs = policies.fingerprint_mols(mols)
+    fprs = policies.fingerprint_mols(mols, expansion_net.X.shape[1])
+
+    if not args.use_openvino:
+        # Predict applicable rules
+        preds = sess.run(expansion_net.pred, feed_dict={
+            expansion_net.keep_prob: 1.,
+            expansion_net.X: fprs,
+            expansion_net.k: 5
+        })
 
-    # Predict applicable rules
-    preds = sess.run(expansion_net.pred_op, feed_dict={
-        expansion_net.keep_prob: 1.,
-        expansion_net.X: fprs,
-        expansion_net.k: 5
-    })
+        indices = preds.indices[0]
+        values = preds.values[0]
+        np.save('expansion_preds_tf.npy', [indices, values])
 
+    else:
+        # Get output nodes names
+        input_blob = next(iter(net.inputs))
+        values_blob = list(iter(net.outputs))[0]
+        indices_blob = list(iter(net.outputs))[1]
+
+        # Predict applicable rules
+        preds = exec_net.infer(inputs={input_blob: fprs})
```

Comment:
Can we use `expansion_net.X` to get input name?

Reply:
`expansion_net.X.name` is `Placeholder:0`, and `input_blob` must be just `Placeholder`, so I should write: 
```
input_blob = expansion_net.X.name.split(':')[0]
```
Is it better?

Reply:
Yes, that's fine!